tpls/mdspan/include/experimental/__p2642_bits/layout_padded.hpp

```diff
@@ -59,8 +59,8 @@ MDSPAN_INLINE_FUNCTION constexpr size_t get_actual_static_padding_value() {
   } else {
     return dynamic_extent;
   }
-  // Missing return statement warning from NVCC and ICC
-#if defined(__NVCC__) || defined(__INTEL_COMPILER)
+  // Missing return statement warning from ICC
+#if defined(__INTEL_COMPILER)
```

Comment:
This may be naive to think we can remove for nvcc. But in src code we seem to only need these returns for intel compiler. E.g. https://github.com/kokkos/kokkos/blob/b367a71b8e9f7d4f1c89231f5b4c6ef09ee77626/core/src/Kokkos_View.hpp#L977

Something that could be related is 
https://github.com/kokkos/kokkos/blob/b367a71b8e9f7d4f1c89231f5b4c6ef09ee77626/tpls/mdspan/include/experimental/__p2630_bits/submdspan_mapping.hpp#L159 
but but I'm not sure how to trigger this setup (when do we have no CUDA_ARCH and nvcc)?

Reply:
Just disable the check when compiling with NVHPC, i.e, when `__NVCOMPILER` is defined.

Reply:
> when do we have no CUDA_ARCH and nvcc?

If nvcc is compiling the host path.

Reply:
> This may be naive to think we can remove for nvcc.

We definitely need this for older `nvcc` versions.